File: STLAF.API/Departments/IT/Models/Ticket.cs
using STLAF.API.Enums;
using STLAF.API.Models;
public class Ticket
{
    public Guid TicketId { get; set; }

    public string TicketNumber { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string CompanyEmail { get; set; } = string.Empty;

    public string ViberNumber { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public TicketCategory Category { get; set; }

    public TicketPriority Priority { get; set; }

    public TicketStatus Status { get; set; } = TicketStatus.Open;

    public Department Department { get; set; }

    public Guid? AssignedTo { get; set; }

    public User? AssignedUser { get; set; }

    public DateTime DateSubmitted { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedDate { get; set; } = DateTime.UtcNow;
}

File: stlaf.frontend/src/departments/IT/types/ticket.ts
export interface TicketDto {
  ticketId: string;
  ticketNumber: string;
  name: string;
  companyEmail: string;
  viberNumber: string;
  description: string;
  category: string;
  priority: string;
  status: string;
  assignedToId?: string | null;
  assignedTo?: string | null;
  dateSubmitted: string;
}

export interface PublicTicketDto {
  ticketNumber: string;
  dateSubmitted: string;
  department: string;
  description: string;
  status: string;
  priority: string;
}

export interface PaginatedResult<T> {
  data: T[];
  totalRecords: number;
  currentPage: number;
  pageSize: number;
  totalPages: number;
}

export interface TicketQuery {
  search?: string;
  status?: string;
  priority?: string;
  assignedTo?: string;
  sortBy?: string;
  descending?: boolean;
  page: number;
  pageSize: number;
}
